pkg/build/api/types.go

```diff
@@ -20,6 +20,30 @@ const (
 	DefaultDockerLabelNamespace = "io.openshift."
 )
 
+const (
+	// SecretTypeSource contains data needed to authenticate against SCM servers.
+	//
+	// SecretTypeSource keys are devided into these option groups:
+	//   Basic authentication group. Requires at least one of fields:
+	//     - Secret.Data["username"] - username used to authenticate against SCM server
+	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
+	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
+	//   SSH authentication group. Required field:
+	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
+	SecretTypeSource kapi.SecretType = "Source"
```

Comment:
> You can't because we have to run against external Kube.  Kube should have
> more secret types and not reject types it doesn't know about.  Kube secret
> types should be more flexible and allow unrelated secret data in them.

Kube doesn't reject unknown secret types for exactly this reason.  Having a way to add additional validation for custom secrets types is strictly better than not having that validation, even if its not always available.

Moving this secret type upstream seemed unlikely given its narrow usage and secrets cannot have multiple types.